src/components/Results.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+import { StyleSheet, View } from 'react-native';
+
+import { colors, normalize, texts } from '../config';
+import { combineLanguages, getAnswerLabel } from '../helpers';
+import { useSurveyLanguages } from '../hooks';
+import { SectionHeader } from './SectionHeader';
+import { ResponseOption } from '../types';
+import { BoldText } from './Text';
+import { Wrapper, WrapperHorizontal, WrapperRow } from './Wrapper';
+
+type Props = { responseOptions: ResponseOption[]; selectedOption: string };
+type SingleProps = {
+  option: ResponseOption & { index: number };
+  selected: boolean;
+  totalCount: number;
+};
+
+const getPercent = (partial: number, total: number) => {
+  if (!total) {
+    return '0%';
+  }
+  return `${((100 * partial) / total).toFixed(0)}%`;
+};
+
+const getCountLabel = (partial: number, total: number) => {
+  return `${partial} / ${getPercent(partial, total)}`;
+};
+
+const SingleResult = ({ option, selected, totalCount }: SingleProps) => {
+  const percentString = getPercent(option.votesCount, totalCount);
+  const width = percentString === '0%' ? barBorderRadius * 2 : percentString;
+  return (
+    <WrapperRow spaceBetween style={styles.container}>
+      <View style={styles.labelContainer}>
+        <BoldText small>{getAnswerLabel('de', option.index)}</BoldText>
+        <BoldText small>{getAnswerLabel('pl', option.index)}</BoldText>
+      </View>
+      <WrapperHorizontal style={styles.barContainer}>
+        <View
+          style={[
+            styles.baseBarStyle,
+            // eslint-disable-next-line react-native/no-inline-styles
+            {
+              opacity: selected ? 1 : 0.5,
+              width
+            }
+          ]}
+        />
+      </WrapperHorizontal>
+      <View style={styles.countContainer}>
+        <BoldText>{getCountLabel(option.votesCount, totalCount)}</BoldText>
+      </View>
+    </WrapperRow>
+  );
+};
+
+export const Results = ({ responseOptions, selectedOption }: Props) => {
+  const languages = useSurveyLanguages();
+  const sortedOptions = responseOptions
+    .map((option, index) => ({ ...option, index }))
+    .sort((a, b) => -a.votesCount + b.votesCount);
```

Comment:
> > ## DeepCode's analysis on [#a1eb19](https://www.deepcode.ai/app/gh/ikuseiGmbH/smart-village-app-app/d7578f0c19ef020691b5aa5503cdebcc18b9cda7/ikuseiGmbH/smart-village-app-app/a1eb1981bd040d991a7d3efb3b7123308c4ee060/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) found:
> > 
> > * ⚠️ **1** warning 👇
> > 
> > ## Top issues
> > Description	Example fixes
> > The callback provided to sort should return 0 if the compared values are equal. Occurrences:
> > 
> > * [Results.tsx:62](https://github.com/ikuseiGmbH/smart-village-app-app/blob/a1eb1981bd040d991a7d3efb3b7123308c4ee060/src/components/Results.tsx#L62)
> > 
> > [🔧 Example fixes](https://www.deepcode.ai/app/gh/ikuseiGmbH/smart-village-app-app/d7578f0c19ef020691b5aa5503cdebcc18b9cda7/ikuseiGmbH/smart-village-app-app/a1eb1981bd040d991a7d3efb3b7123308c4ee060/pr/_/%2Fsrc%2Fcomponents%2FResults.tsx/javascript%2Fdc_interfile_project%2FNoZeroReturnedInSort/code/?utm_source=gh_review&)
> > #### 👉 View analysis in [**DeepCode’s Dashboard**](https://www.deepcode.ai/app/gh/ikuseiGmbH/smart-village-app-app/d7578f0c19ef020691b5aa5503cdebcc18b9cda7/ikuseiGmbH/smart-village-app-app/a1eb1981bd040d991a7d3efb3b7123308c4ee060/pr/_/%2F/code/?utm_source=gh_review&c=0&w=1&i=0&) | [_Configure the bot_](https://www.deepcode.ai/app/gh/?ownerconfig=ikuseiGmbH)
> > #### 👉 The DeepCode service and API will be deprecated in August, 2021. [Here](https://www.deepcode.ai/blog/migrate-to-snyk-code) is the information how to migrate. Thank you for using DeepCode 🙏 ❤️ !
> > If you are using our plugins, you might be interested in their successors: [Snyk's JetBrains plugin](https://snyk.co/udpkq) and [Snyk's VS Code plugin](https://snyk.co/udpkr).
> 
> it does?

i think deepcode cannot see, that we have to numbers being compared. for us it is clear that -5 + 5 is 0. i rhink deepcode requires a sort function with explicit cases for <0, 0 or >0. 

we can add an ignore for this:

```
// deepcode ignore NoZeroReturnedInSort: The callback provided to sort does return 0 implicitly if the compared values are equal.
.sort((a, b) => -a.votesCount + b.votesCount);
```

what do you say?

Reply:
added an ignore comment